src/test/java/ArrayIteratorTest.java

```diff
@@ -0,0 +1,129 @@
+import org.junit.jupiter.api.Test;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ArrayIteratorTest {
+
+    @Test
+    void hasNext_emptyArray_returnsFalse() {
+        Integer[] a = {null, null, null, null, null, null, null, null, null, null};
+        Iterator<Integer> it = new ArrayIterator<>(a, 0);
+        assertFalse(it.hasNext());
+    }
+
+    @Test
+    void hasNext_singleElementArray_returnsTrue() {
+        Integer[] a = {10, null, null, null, null, null, null, null, null, null};
+        Iterator<Integer> it = new ArrayIterator<>(a, 1);
+        assertTrue(it.hasNext());
+    }
+
+    @Test
+    void hasNext_multiElementArray_returnsTrue() {
+        Integer[] a = {10, 11, 12, 13, 14, null, null, null, null, null};
+        Iterator<Integer> it = new ArrayIterator<>(a, 5);
+        assertTrue(it.hasNext());
+    }
+
+    @Test
+    void hasNext_singleElementArrayAfterHasNextAndNoNextCall_returnsTrue() {
+        Integer[] a = {10, null, null, null, null, null, null, null, null, null};
+        Iterator<Integer> it = new ArrayIterator<>(a, 1);
+        it.hasNext();
+        assertTrue(it.hasNext());
+    }
+
+    @Test
+    void hasNext_multiElementArrayAfterHasNextAndNoNextCall_returnsTrue() {
+        Integer[] a = {10, 11, 12, 13, 14, null, null, null, null, null};
+        Iterator<Integer> it = new ArrayIterator<>(a, 5);
+        it.hasNext();
+        it.hasNext();
+        it.hasNext();
+        it.hasNext();
+        it.hasNext();
+        assertTrue(it.hasNext());
+    }
+
+    @Test
+    void hasNext_endOfSingleElementArray_returnsFalse() {
+        Integer[] a = {10, null, null, null, null, null, null, null, null, null};
+        Iterator<Integer> it = new ArrayIterator<>(a, 1);
+        it.next();
```

Comment:
Shouldn't this throw? `1` is the "end" of the array.

Reply:
Not sure what you mean. Since the "`size`" is `1`, calling `next` once will move the iterator down one and result in it being "finished", for lack of a better term.

Reply:
Ah, I misinterpreted. 1 is number of elements, not like a starting index.